org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenCompletionParticipant.java

```diff
@@ -253,6 +264,58 @@ private void completeProperties(ICompletionRequest request, ICompletionResponse
 		}
 	}
 
+	private void collectVersionCompletion(ICompletionRequest request, ICompletionResponse response) {
+		DOMElement node = request.getParentElement();
+		DOMDocument doc = request.getXMLDocument();
+
+		Range range = XMLPositionUtility.createRange(node.getStartTagCloseOffset() + 1, node.getEndTagOpenOffset(),
+				doc);
+		ArtifactVersionSearcher artifactVersionSearcher = ArtifactVersionSearcher.getInstance();
+
+		if (indexSyncRequest == null) {
+			try {
+				indexSyncRequest = artifactVersionSearcher.init(cache.getPlexusContainer(),

```

Comment:
@mickaelistria I moved all of the completableFuture.get() and .isDone() calls out of ArtifactVersionSearcher. I think this is looking better, as code specific to version completion (eg. error responses) should be handled by MavenCompletionParticpant, as ArtifactVersionSearcher will eventually supported GroupId and VersionId searching.